app/assets/js/services/basket.js

```diff
@@ -0,0 +1,41 @@
+function BasketService ($http, $q, $log, API, universal_variable, browserStorage) {
+	var service = {},
+		basket_id = browserStorage.getItem('basket_id');
+
+	service.createBasket = function() {
+		return $http({
+			method: 'POST',
+			url: API.baskets
+		})
+			.then(function(basket) {
+				basket_id = basket.id;
+				browserStorage.setItem('basket_id', basket_id);
+				universal_variable.basket = basket;
+				return basket;
+			});
+	};
+
+	service.getBasket = function() {
+		if (!basket_id || browserStorage.getItem('basket_id')) {
+			return service.createBasket();
+		}
+
+		return $http({
+			method: 'GET',
+			url: API.baskets + '/' + basket_id
+		})
+			.then(function(basket) {
+				universal_variable.basket = basket;
+				return basket;
+			});
+	};
+
+	service
```

Comment:
`getBasket()` suggests that this function just gets a basket but I think it's being used for side effects here. Might be able to move the part that checks if the basket exists to `createBasket()` or extract to a separate method?